producer.go

```diff
@@ -136,6 +136,10 @@ func extendNameWithRandSuffix(name string) (string, error) {
 
 // CreateProducer - creates a producer.
 func (c *Conn) CreateProducer(stationName, name string, opts ...ProducerOpt) (*Producer, error) {
+	if err, p := checkForProducerExistence(stationName, name); err == nil {
```

Comment:
I'd like to leave the current createProducer implementation as is, and move this logic into the new function you created

Reply:
Why wouldn't we want to check if it already exists to not waste time while creating a new producer(only because we have this new command of Produce from connection)?